src/filter/TargetingFilter.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { IFeatureFilter } from "./FeatureFilter";
+import { createHash } from "crypto";
+
+type TargetingFilterParameters = {
+    Audience: {
+        DefaultRolloutPercentage: number;
+        Users?: string[];
+        Groups?: {
+            Name: string;
+            RolloutPercentage: number;
+        }[];
+        Exclusion?: {
+            Users?: string[];
+            Groups?: string[];
+        };
+    }
+}
+
+type TargetingFilterEvaluationContext = {
+    featureName: string;
+    parameters: TargetingFilterParameters;
+}
+
+type TargetingFilterAppContext = {
+    userId?: string;
+    groups?: string[];
+}
+
+export class TargetingFilter implements IFeatureFilter {
+    name: string = "Microsoft.Targeting";
+
+    evaluate(context: TargetingFilterEvaluationContext, appContext?: TargetingFilterAppContext): boolean | Promise<boolean> {

```

Comment:
This feels like a lie- since we don't ever return Promise<boolean>. I prefer marking this async and only doing Promise.

Reply:
@rossgrambo Good catch. That was a miss. 
According to current impl, it returns a boolean instead of a promise, without any async operations. context is read from feature flags already loaded, and appContext is provided directly in parameter. 
I'll create a PR changing it to 
```js
    evaluate(context: TargetingFilterEvaluationContext, appContext?: TargetingFilterAppContext): boolean
```

WDYT?